src/Dfe.PlanTech.Application/Users/Commands/CreateEstablishmentCommand.cs

```diff
@@ -0,0 +1,40 @@
+using Dfe.PlanTech.Application.Persistence.Interfaces;
+using Dfe.PlanTech.Domain.Establishments.Models;
+
+namespace Dfe.PlanTech.Application.Users.Commands;
+
+public class CreateEstablishmentCommand : ICreateEstablishmentCommand
+{
+    private readonly IPlanTechDbContext _db;
+
+    public CreateEstablishmentCommand(IPlanTechDbContext db)
+    {
+        _db = db;
+    }
+
+    /// <summary>
+    /// Creates new user and returns ID
+    /// </summary>
+    /// <param name="establishmentDto"></param>
+    /// <returns></returns>
+    public async Task<int> CreateEstablishment(EstablishmentDto establishmentDto)
+    {
+        if (establishmentDto.Urn == null && establishmentDto.Ukprn == null)
+        {
+            throw new InvalidOperationException("Both Urn and Ukprn cannot be null.");
+        }
+        
+        var establishment = new Establishment()
+        {
+            EstablishmentRef = establishmentDto.Urn != null ? establishmentDto.Urn : establishmentDto.Ukprn!,
```

Comment:
Purely stylistic, but you could also do:

`EstablishmentRef = establishmentDto?.Urn ?? establishmentDto.Ukprn!`

Only mentioning this as I'm pretty sure this doesn't exist in Java, so thought you might be interested if unaware!

Again, purely stylistic, so up to you :)